frontend/src/utils/adequacy.ts

```diff
@@ -0,0 +1,72 @@
+import { PostAdequaciesResponse } from '../constants/api/adequacies-response'
+import { AdequacyMode, Method } from '../constants/datatypes'
+
+let ONE_MILE_IN_METERS = 1609.344
+let ONE_METER_IN_MILES = 1.0 / ONE_MILE_IN_METERS
+
+export function getAdequacyMode(
+  adequacy: PostAdequaciesResponse[0],
+  method: Method,
+  serviceAreaId: string,
+  selectedServiceAreas: string[] | null
+): AdequacyMode {
+
+  if (selectedServiceAreas && !selectedServiceAreas.includes(serviceAreaId)) {
+    return AdequacyMode.OUT_OF_SCOPE
+  }
+
+  if (method === 'straight_line') {
+    if (adequacy.to_closest_provider * ONE_METER_IN_MILES <= 10) {
+      return AdequacyMode.ADEQUATE_0
+    }
+    if (adequacy.to_closest_provider * ONE_METER_IN_MILES <= 20) {
+      return AdequacyMode.ADEQUATE_1
+    }
+    if (adequacy.to_closest_provider * ONE_METER_IN_MILES <= 30) {
+      return AdequacyMode.ADEQUATE_2
+    }
+    if (adequacy.to_closest_provider * ONE_METER_IN_MILES > 30) {
+      return AdequacyMode.INADEQUATE
+    }
+    return AdequacyMode.OUT_OF_SCOPE
+  }
+
+  if (method === 'driving_time' || method === 'walking_time') {
+    if (adequacy.to_closest_provider <= 30) {
+      return AdequacyMode.ADEQUATE_0
+    }
+    if (adequacy.to_closest_provider <= 45) {
+      return AdequacyMode.ADEQUATE_1
+    }
+    if (adequacy.to_closest_provider <= 60) {
+      return AdequacyMode.ADEQUATE_2
+    }
+    if (adequacy.to_closest_provider > 60) {
+      return AdequacyMode.INADEQUATE
+    }
+  }
+  return AdequacyMode.OUT_OF_SCOPE
+}
+
+export function getLegend(method: Method, standard: AdequacyMode) {
```

Comment:
seems like there are two of these. very confused

Reply:
also not sure that I'd look for this function in adequacy. not sure where else would be good though